server/api/user.go

```diff
@@ -70,20 +70,23 @@ func GetRepos(c *gin.Context) {
 	}
 
 	if all {
-		active := map[string]bool{}
+		active := map[model.ForgeRemoteID]*model.Repo{}
 		for _, r := range activeRepos {
-			active[r.FullName] = r.IsActive
+			active[r.ForgeRemoteID] = r
 		}
 
 		_repos, err := _forge.Repos(c, user)
 		if err != nil {
 			c.String(http.StatusInternalServerError, "Error fetching repository list. %s", err)
 			return
 		}
+
 		var repos []*model.Repo
 		for _, r := range _repos {
 			if r.Perm.Push {
-				if active[r.FullName] {
+				if active[r.ForgeRemoteID] != nil && active[r.ForgeRemoteID].IsActive {
+					// TODO: should we merge other fields as well?
```

Comment:
@qwerty287 I had to change this matching a bit to be independent of the full-name and copied the existing id in case the repo already exists. Should we merge all other "internal"  values which are not set by the remote as well?

Reply:
@qwerty287 Changed it to use the existing repo entry with updated values using our existing `repo.Update(from)` function